sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -86,6 +86,7 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
             if (msg.content.value == _TIMEOUT_MAGIC) {
                 return -ETIMEDOUT;
             }
+            __attribute__((fallthrough));
```

Comment:
>i don't think so, it think it must be above the case it falls into.

but in your provided examples in https://github.com/RIOT-OS/RIOT/pull/7403#discussion_r129269059 you didn't try to put the comment above of the case.